municipal_finance/summarise_data.py

```diff
@@ -0,0 +1,119 @@
+from django.db.models import Min, Max
+from django_q.tasks import async_task
+from django.db import transaction
+
+from municipal_finance.models.data_summaries import Summary
+
+from scorecard.models.geography import Geography
+
+from municipal_finance.models.cash_flow import (
+    CflowFactsV1,
+    CflowFactsV2,
+)
+from municipal_finance.models.income_expenditure import (
+    IncexpFactsV1,
+    IncexpFactsV2,
+)
+from municipal_finance.models.financial_position import (
+    BsheetFactsV1,
+    FinancialPositionFactsV2,
+)
+from municipal_finance.models.capital import (
+    CapitalFactsV1,
+    CapitalFactsV2,
+)
+from municipal_finance.models.grants import (
+    ConditionalGrantFactsV1,
+    GrantFactsV2,
+)
+from municipal_finance.models.repairs_maintenance import (
+    RepairsMaintenanceFactsV1,
+    RepairsMaintenanceFactsV2,
+)
+from municipal_finance.models.aged_debtor import (
+    AgedDebtorFactsV1,
+    AgedDebtorFactsV2,
+)
+from municipal_finance.models.aged_creditor import (
+    AgedCreditorFactsV1,
+    AgedCreditorFactsV2,
+)
+from municipal_finance.models.uifw_expense import UIFWExpenseFacts
+
+
+FACT_TABLES = [
+    CflowFactsV1,
+    CflowFactsV2,
+    IncexpFactsV1,
+    IncexpFactsV2,
+    BsheetFactsV1,
+    FinancialPositionFactsV2,
+    CapitalFactsV1,
+    CapitalFactsV2,
+    ConditionalGrantFactsV1,
+    GrantFactsV2,
+    RepairsMaintenanceFactsV1,
+    RepairsMaintenanceFactsV2,
+    AgedDebtorFactsV1,
+    AgedDebtorFactsV2,
+    AgedCreditorFactsV1,
+    AgedCreditorFactsV2,
+    UIFWExpenseFacts
+]
+
+
+def summarise_task(task):
+    if task.success:
+        async_task(
+            "municipal_finance.summarise_data.summarise",
+            task_name="Summarise Data",
+        )
+
+
+@transaction.atomic
+def summarise():
+    min_year = 3000
+    max_year = 1000
+    count_facts = 0
+
+    for table in FACT_TABLES:
+        min_year = get_min(min_year, table)
+        max_year = get_max(max_year, table)
+
+    count_years = max_year - min_year
+    Summary.objects.update_or_create(
+        type='years',
+        defaults={
+            'content': f'{{"count":{count_years}, "min":{min_year}, "max":{max_year}}}'}
+    )
+
+    total = Geography.objects.all().count()
+    metros = Geography.objects.filter(category='A').count()
+    munis = Geography.objects.filter(category='B').count()
+    districts = Geography.objects.filter(category='C').count()
```

Comment:
this needs to exclude disestablished municipalities https://municipaldata.treasury.gov.za/api/cubes/demarcation_changes/facts

Reply:
otherwise you'll be over-counting

Reply:
Discussed in standup - added a note making it clear that disestablished municipalities are included in the data

Reply:
nice - I don't think `sup` is the right thing - I think it needs space below the big text, and a matching *. The size of the text is fine, so just style it another way to get smaller text and a bit more space

![2023-01-11_11-46](https://user-images.githubusercontent.com/235801/211772994-2ea45cc7-8c45-4a9d-ad9b-79687fa90fc7.png)

Reply:
Changed to <p>, added matching asterisk and extra space